packages/select/src/components/query-list/queryList.tsx

```diff
@@ -257,18 +267,39 @@ export class QueryList<T> extends React.Component<IQueryListProps<T>, IQueryList
         });
     };
 
+    private renderCreateItemMenuItem = (query: string) => {
+        const handleClick: React.MouseEventHandler<HTMLElement> = evt => {
+            this.handleItemCreate(query, evt);
+        };
+        return Utils.safeInvoke(
+            this.props.createNewItemRenderer,
+            query,
+            this.state.activeItem ? this.state.activeItem.type === QueryListActiveItemType.CREATE : false,
```

Comment:
- Unpack since you use `this.state.activeItem` twice.
- Use explicit boolean expression (`activeItem != null`).